src/org/zaproxy/zap/extension/openapi/converter/swagger/SwaggerConverter.java

```diff
@@ -74,6 +79,25 @@ public SwaggerConverter(Scheme defaultScheme, String defn, ValueGenerator valueG
     private List<OperationModel> readOpenAPISpec() throws SwaggerException {
         List<OperationModel> operations = new LinkedList<>();
         Swagger swagger = new SwaggerParser().parse(this.defn);
+        
+        if (swagger == null) {
+            try {
+                // Try the older spec
+                // Annoyingly the converter only reads files
+                File temp = File.createTempFile("openapi", ".defn");
+                BufferedWriter bw = new BufferedWriter(new FileWriter(temp));
+                bw.write(this.defn);
+                bw.close();
+                
+                swagger = new SwaggerCompatConverter().read(temp.getAbsolutePath());
+                if (!temp.delete()) {
+                    LOG.debug("Failed to delete " + temp.getAbsolutePath());

```

Comment:
tbh this was more to avoid a Coverity warning ;) I'd hope that if we can create temp files then we can also delete them. But I'm fine changing it to a warning